SDL_ttf.c

```diff
@@ -3128,6 +3128,52 @@ int TTF_SetFontScriptName(TTF_Font *font, const char *script)
 #endif
 }
 
+extern DECLSPEC int SDLCALL TTF_GetScript(Uint32 ch, char* script)
+{
+#if TTF_USE_HARFBUZZ
+
+    hb_buffer_t* hb_buffer = hb_buffer_create();
+
+    if (hb_buffer == NULL)
+    {
+        TTF_SetError("Cannot create harfbuzz buffer");
+        return -1;
+    }
+
+    if (script == NULL || SDL_strlen(script) != 4)
+    {
+        return -1;
+    }
+
+    hb_unicode_funcs_t* hb_unicode_functions = hb_buffer_get_unicode_funcs(hb_buffer);
+
+    if (hb_unicode_functions == NULL)
+    {
+        TTF_SetError("Cannot get harfbuzz unicode funcs");
+
+        hb_buffer_destroy(hb_buffer);
+        return -1;
+    }
+
+    hb_buffer_clear_contents(hb_buffer);
+    hb_buffer_set_content_type(hb_buffer, HB_BUFFER_CONTENT_TYPE_UNICODE);
+
+    uint8_t const untagged_script[4] = { HB_UNTAG(hb_unicode_script(hb_unicode_functions, ch)) };
+
+    script[0] = (char)untagged_script[0];
+    script[1] = (char)untagged_script[1];
+    script[2] = (char)untagged_script[2];
+    script[3] = (char)untagged_script[3];
```

Comment:
Do we not want nul terminating `script` here?

Reply:
Resolved